apps/user/permissions.py

```diff
@@ -2,15 +2,16 @@
 
 
 class UserPermission(permissions.BasePermission):
-    def has_object_permission(self, request, view, obj):
-        if request.method in permissions.SAFE_METHODS:
-            return True
-        return obj == request.user
-
+    def _is_authenticated(self, rq):
+        return rq.user and rq.user.is_authenticated
```

Comment:
Not the most important thing.

`request.user` is always going to be either `User` instance or `contrib.auth.AnonymousUser` instance.

```suggestion
    def _is_authenticated(self, rq):
        return rq.user.is_authenticated
```